### PR TITLE
Add unit tests for retry, nodepool, and config packages

### LINKED ISSUE #4: Expand unit test coverage beyond util package
## Summary

Currently only `internal/util/amount_test.go` has tests. Key packages would benefit from unit test coverage.

## Priority Packages

1. **`internal/retry`** — straightforward to test with mock functions, verify backoff timing and retryable error detection
2. **`internal/nodepool`** — failover/recover logic, `FallbackClient()`, `CheckHealth()` primary recovery
3. **`internal/health`** — cache TTL behavior, RWMutex correctness, degraded status on node errors
4. **`internal/config`** — flag parsing, network presets (mainnet/nile/shasta), env var overrides

## Approach

- Use table-driven tests per project conventions
- Mock gRPC clients for nodepool and health tests
- Test edge cases: MinInt64 in SunToTRX, zero decimals in FormatTRC20Amount, concurrent access to health cache

### PR DESCRIPTION
**Pull request by fbsobreira**
## Summary

Expands test coverage from 1 package (util) to 4 packages:

- **`internal/retry`** — 6 tests: success on first try, success after retries, exhausted retries, non-retryable errors, retryable error detection (table-driven), backoff timing validation
- **`internal/nodepool`** — 12 tests: Client/ActiveNode/ClientAndNode, FallbackClient (nil + configured), Failover (success, no fallback, already on fallback), Recover (success, already on primary), full failover-recover cycle, ClientAndNode after failover
- **`internal/config`** — 4 tests: IsHostedMode, envOrDefault with env set/empty/unset, network node presets, unknown network fallback

Health package tests deferred — requires interface extraction to mock gRPC calls in `check()`.

Closes #4

## Test plan

- [x] `make fmt` — no changes
- [x] `make lint` — 0 issues
- [x] `make test` — all 22 new tests pass with `-race`

## Summary by CodeRabbit

* **Tests**
  * Added unit tests for configuration behavior: hosted-mode detection, environment-variable fallback logic, and network endpoint mapping with unknown-network fallback.
  * Added a comprehensive node-pool test suite for primary/fallback selection, failover and recovery flows, state consistency, and concurrent failover/recover stress testing.
  * Added retry-utility tests covering retryable vs non-retryable errors, exhausted retries, success-after-retries, and backoff timing.

### CODE REVIEW

File: internal/nodepool/pool_test.go [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+	p.Failover()
+
+	c, addr := p.ClientAndNode()
+	if c != p.fallback.client {
+		t.Error("ClientAndNode() client should be fallback after failover")
+	}
+	if addr != "fallback:50051" {
+		t.Errorf("ClientAndNode() addr = %q, want %q", addr, "fallback:50051")
+	}
+}
```
[the remark this answers is not in the document]
Comment (fbsobreira) [reply internal/nodepool/pool_test.go]: Fixed — added `TestConcurrentFailoverRecover` with 10 goroutines × 100 iterations each, calling Failover/Recover concurrently. Asserts invariants on ClientAndNode snapshot consistency after completion. Runs with `-race`.

File: internal/config/config_test.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+	}{
+		{"env set", "TEST_CONFIG_VAR", "from-env", true, "default", "from-env"},
+		{"env empty uses fallback", "TEST_CONFIG_EMPTY", "", true, "default", "default"},
+		{"env unset uses fallback", "TEST_CONFIG_UNSET_XYZ", "", false, "fallback-val", "fallback-val"},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if tt.setEnv {
+				t.Setenv(tt.key, tt.envVal)
+			}
```
[the remark this answers is not in the document]
Comment (fbsobreira) [reply internal/config/config_test.go]: Fixed — added explicit `os.Unsetenv(tt.key)` in the `!setEnv` branch to guarantee the variable is absent regardless of the runner environment.
